GiniVision/Classes/Core/Screens/Camera/CameraPreviewViewController.swift

```diff
@@ -0,0 +1,292 @@
+    fileprivate var cameraState = CameraState.notValid
+    fileprivate var defaultImageView: UIImageView?
+    fileprivate var focusIndicatorImageView: UIImageView?
+    fileprivate let interfaceOrientationsMapping: [UIInterfaceOrientation: AVCaptureVideoOrientation] = [
+        .portrait: .portrait,
+        .landscapeRight: .landscapeRight,
+        .landscapeLeft: .landscapeLeft,
+        .portraitUpsideDown: .portraitUpsideDown
+    ]
+    
+    fileprivate var cameraFocusSmall: UIImage? {
+        return UIImageNamedPreferred(named: "cameraFocusSmall")
+    }
+    
+    fileprivate var cameraFocusLarge: UIImage? {
+        return UIImageNamedPreferred(named: "cameraFocusLarge")
+    }
+    
+    fileprivate var defaultImage: UIImage? {
+        return UIImageNamedPreferred(named: "cameraDefaultDocumentImage")
+    }
+    
+    fileprivate enum CameraState {
+        case valid, notValid
+    }
+
+    lazy var previewView: CameraPreviewView = {
+        let previewView = CameraPreviewView()
+        previewView.translatesAutoresizingMaskIntoConstraints = false
+        (previewView.layer as? AVCaptureVideoPreviewLayer)?.videoGravity = .resize
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(focusAndExposeTap))
+        previewView.addGestureRecognizer(tapGesture)
+        return previewView
+    }()
+    
+    init(giniConfiguration: GiniConfiguration = .shared) {
+        self.giniConfiguration = giniConfiguration
+        super.init(nibName: nil, bundle: nil)
+        
+        setupCamera(giniConfiguration: giniConfiguration)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    override func loadView() {
+        super.loadView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        
+        if let validCamera = camera {
+            cameraState = .valid
+            previewView.session = validCamera.session
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(subjectAreaDidChange),
+                                                   name: NSNotification.Name.AVCaptureDeviceSubjectAreaDidChange,
+                                                   object: camera?.videoDeviceInput?.device)
+        }
+        previewView.drawGuides(withColor: giniConfiguration.cameraPreviewCornerGuidesColor)
+        
+        view.insertSubview(previewView, at: 0)
+        Constraints.pin(view: previewView, toSuperView: view)
+
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        updatePreviewViewOrientation() // Video orientation should be updated once the view has been loaded
+    }
+    
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        camera?.start()
```

Comment:
Can you make camera non-optional? Force-unwrapped variable might also be fine. If the camera turns out to be nil, the app better crash. Just IMHO.

Reply:
Definitely makes sense, I just copied and paster most of the stuff from one file to the other and didn't consider making many changes tbh.